hera_qm/xrfi.py

```diff
@@ -711,40 +829,90 @@ def flag(uvf_m, nsig_p=6., nsig_f=None, nsig_t=None, avg_method='quadmean'):
             uvf_f.flag_array[:, :, indf, :, :] = True
         if nsig_t is not None:
             # Time watershed
             data = uvutils.collapse(uvf_m.metric_array, avg_method, axis=(0, 1, 2, 4),
                                     weights=uvf_m.weights_array)
-            indt = np.where(data >= nsig_t)[0]
+            indt = np.where(np.abs(data) >= nsig_t)[0]
             uvf_f.flag_array[:, :, :, indt, :] = True
     elif uvf_m.type == 'waterfall':
         if nsig_f is not None:
             # Channel flag
             data = uvutils.collapse(uvf_m.metric_array, avg_method, axis=(0, 2),
                                     weights=uvf_m.weights_array)
-            indf = np.where(data >= nsig_f)[0]
+            indf = np.where(np.abs(data) >= nsig_f)[0]
             uvf_f.flag_array[:, indf, :] = True
         if nsig_t is not None:
             # Time watershed
             data = uvutils.collapse(uvf_m.metric_array, avg_method, axis=(1, 2),
                                     weights=uvf_m.weights_array)
-            indt = np.where(data >= nsig_t)[0]
+            indt = np.where(np.abs(data) >= nsig_t)[0]
             uvf_f.flag_array[indt, :, :] = True
     else:
         raise ValueError('Unknown UVFlag type: ' + uvf_m.type)
     return uvf_f
 
 
+def threshold_wf(uvf_m, nsig_f=5.0, nsig_t=5.0, detrend=True):
+    """Flag on a "waterfall" type UVFlag in "metric" mode.
+
+    Collapses to one dimension, thresholds, then broadcasts back to waterfall.
+
+    Parameters
+    ----------
+    uvf_m : UVFlag object
+        A UVFlag object in 'metric' mode (i.e., number of sigma data is from middle),
+        and type 'waterfall'.
+    nsig_f : float, optional
+        The number of sigma above which to flag channels. Default is 5.0.
+    nsig_t : float, optional
+        The number of sigma above which to flag integrations. Default is 5.0.
+    detrend : bool, optional
+        Whether to detrend the 1D data before calculating zscores. Default is True.
+
+    Returns
+    -------
+    uvf_f : UVFlag object
+        A UVFlag object in 'flag' mode with flags determined from uvf_m.
+
+    Raises
+    ------
+    ValueError:
+        If uvf_m is not a UVFlag object in metric mode, or if the type of uvf_m
+        is not "waterfall", a ValueError is raised.
+    """
+    # Check input
+    if (not isinstance(uvf_m, UVFlag)) or (uvf_m.mode != 'metric') or (uvf_m.type != 'waterfall'):
+        raise ValueError('uvf_m must be UVFlag instance with mode == "metric" and'
+                         + ' type == "waterfall."')
+
+    # initialize
+    uvf_f = uvf_m.copy()
+    uvf_f.to_flag()
+    # Mask invalid values. Masked medians are slow, but these are done only on waterfalls.
+    data = np.ma.masked_array(uvf_m.metric_array, ~np.isfinite(uvf_m.metric_array))
+    # Collapse to 1D and calculate z scores
+    spec = np.ma.median(data, axis=(0, 2))
+    zspec = modzscore_1d(spec, detrend=detrend)
```

Comment:
`np.ma.median` does, but unfortunately the medfilt in `modzscore_1d` does not. I will document this.
In practice, the masked median is useful because we have large chunks of data missing because the sun is up and the band edges that are flagged. After the median above, the missing data _should_ be limited to the edges (in either time or freq)

Reply:
Sounds good.